crates/shared/src/order_validation.rs

```diff
@@ -335,16 +330,6 @@ impl OrderValidating for OrderValidator {
         if order.sell_token == BUY_ETH_ADDRESS {
             return Err(PartialValidationError::InvalidNativeSellToken);
         }
-        if order.buy_token == BUY_ETH_ADDRESS {
```

Comment:
Removing this check is dangerous as it assumes that we can handle the access list generation in the driver. Note that it is possible for the driver to be configured without access list creation at the moment, so we should probably have a way to stop accepting smart contract `BUY_ETH` orders in case we can't support access list creation in the driver.

I had a draft PR for this flag a while back (https://github.com/cowprotocol/services/pull/139). I think we should feature-gate accepting `BUY_ETH` orders from smart contracts for now.